DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserDataService.cs

```diff
@@ -65,6 +66,92 @@ public partial class UserDataService
+                da.CandidateNumber,
+                LTRIM(RTRIM(da.Answer1)) AS Answer1,
+                LTRIM(RTRIM(da.Answer2)) AS Answer2,
+                LTRIM(RTRIM(da.Answer3)) AS Answer3,
+                LTRIM(RTRIM(da.Answer4)) AS Answer4,
+                LTRIM(RTRIM(da.Answer5)) AS Answer5,
+                LTRIM(RTRIM(da.Answer6)) AS Answer6,
+                da.Approved,
+                da.ExternalReg,
+                da.SelfReg,
+                da.OldPassword,
+                da.UserID,
+                da.CentreSpecificDetailsLastChecked,
+                u.ID,
+                u.PrimaryEmail,
+                u.PasswordHash,
+                u.FirstName,
+                u.LastName,
+                u.JobGroupID,
+                jg.JobGroupName,
+                u.ProfessionalRegistrationNumber,
+                u.ProfileImage,
+                u.Active,
+                u.ResetPasswordID,
+                u.TermsAgreed,
+                u.FailedLoginCount,
+                u.HasBeenPromptedForPrn,
+                u.LearningHubAuthID,
+                u.HasDismissedLhLoginWarning,
+                u.EmailVerified,
+                u.DetailsLastChecked,
+                CASE
+                    WHEN ucd.ID IS NOT NULL THEN ucd.ID
+                    ELSE 0
+                END AS ID,
+                CASE
+                    WHEN ucd.UserID IS NOT NULL THEN ucd.UserID
+                    ELSE 0
+                END AS UserID,
+                CASE
+                    WHEN ucd.CentreID IS NOT NULL THEN ucd.CentreID
+                    ELSE 0
+                END AS CentreID,
+                CASE
+                    WHEN ucd.Email IS NOT NULL THEN ucd.Email
+                    ELSE NULL
+                END AS Email,
+                CASE
+                    WHEN ucd.EmailVerified IS NOT NULL THEN ucd.EmailVerified
+                    ELSE NULL
+                END AS EmailVerified
+            FROM DelegateAccounts AS da
+            INNER JOIN Centres AS ce ON ce.CentreId = da.CentreID
+            INNER JOIN Users AS u ON u.ID = da.UserID
+            LEFT JOIN UserCentreDetails AS ucd ON ucd.UserID = u.ID
+                AND ucd.CentreId = da.CentreID
+            INNER JOIN JobGroups AS jg ON jg.JobGroupID = u.JobGroupID
+            WHERE da.ID = ${id}";
+
+            var delegateUser = connection.Query<DelegateAccount, UserAccount, UserCentreDetails, Delegate>(
+                sql,
+                (delegateAccount, userAccount, userCentreDetails) =>
+                {
+                    if (userCentreDetails.Id == 0)
+                    {
+                        userCentreDetails = null!;
+                    }
+
+                    var delegateUser = new Delegate(delegateAccount, userAccount, userCentreDetails);
+                    return delegateUser;
+                },
+                splitOn: "ID,ID"
+            ).FirstOrDefault();
+
+            return delegateUser;

```

Comment:
Thanks, made those changes. I'm not sure where I got the trimming stuff from, I know I saw it somewhere else in the repo but can't remember where. But yes, I agree that we shouldn't need to trim the answers here, that should be happening when saving the answers rather than getting them.